src/mavsdk/core/include/mavsdk/component_type.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+namespace mavsdk {
+
+    /**
+     * @brief ComponentType of configurations, used for automatic ID setting
+     */
+    enum class ComponentType {
+        Autopilot, /**< @brief SDK is used as an autopilot. */
+        GroundStation, /**< @brief SDK is used as a ground station. */
+        CompanionComputer, /**< @brief SDK is used as a companion computer on board the MAV. */
+        Camera, /** < @brief SDK is used as a camera. */
+        Gimbal, /** < @brief SDK is used as a gimbal. */
+        Custom /**< @brief the SDK is used in a custom configuration, no automatic ID will be
+                  provided */
+    };
```

Comment:
add `RemoteId`

Reply:
Yes, please make another one.

Reply:
I checked out this PR and added a commit on top. Once this is merged I'll rebase and push again.